omod/src/main/java/org/openmrs/module/smartonfhir/web/servlet/SmartAppSelectorServlet.java

```diff
@@ -15,15 +15,27 @@
 
 import java.io.IOException;
 
+import ca.uhn.fhir.rest.server.IServerAddressStrategy;
 import org.apache.commons.lang3.StringUtils;
 import org.apache.http.HttpStatus;
+import org.springframework.beans.factory.annotation.Autowired;
 
 public class SmartAppSelectorServlet extends HttpServlet {
 	
+	@Autowired
+	private IServerAddressStrategy iServerAddressStrategy;
+	
 	@Override
 	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+		String baseURL = iServerAddressStrategy.determineServerBase(req.getServletContext(), req);
 		String smartAppLaunchURL = req.getParameter("launchUrl");
-		String url = smartAppLaunchURL + "?iss=http://localhost:8080/openmrs/ws/fhir2/R4&launch=";
+		
+		if (!(baseURL.contains("R4") || baseURL.contains("R3"))) {
```

Comment:
What happens if the `fhirVersion` parameter is missing?

Reply:
If the Base URL contains the fhirVersion, then we go with it, but if not then we will set the fhirVersion to R4. I preferred the Base URL fhirVersion (if returned) more than the fhirVersion mentioned in the config file. Thought? 

I think we can encounter one more case if the user doesn't mention fhirVersion in the config file and the Base URL also doesn't contain fhirVersion, setting the fhirVersion to R4 is a good choice. Thought?

Reply:
@theanandankit That all sounds right. I just want to make sure we have some kind of default value in case the request doesn't contain the `fhirVersion` parameter.